subprojects/docs/src/docs/userguide/basePlugin.adoc

```diff
@@ -0,0 +1,90 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[base_plugin]]
+== The Base Plugin
+
+The Base Plugin provides some tasks and conventions that are common to most builds and adds a structure to the build that promotes consistency in how they are run. Its most significant contribution is a set of <<sec:lifecycle_tasks,_lifecycle_ tasks>> that act as an umbrella for the more specific tasks provided by other plugins and build authors.
+
+[[sec:base_plugin_usage]]
+=== Usage
+
+++++
+<sample id="applyingBasePlugin" dir="userguide/basePlugin" title="Applying the Base Plugin">
+    <sourcefile file="build.gradle" snippet="apply-base-plugin"/>
+</sample>
+++++
+
+[[sec:base_tasks]]
+=== Tasks
+
+`clean` — type: `Delete`::
+Deletes the build directory and everything in it, i.e. the path specified by the api:org.gradle.api.Project#getBuildDir()[] project property.
+
+`check` — _lifecycle task_::
+Plugins and build authors should attach their verification tasks, such as ones that run tests, to this lifecycle task using `check.dependsOn(__task__)`.
+
+`assemble` — _lifecycle task_::
+Plugins and build authors should attach tasks that produce distributions and other consumable artifacts to this lifecycle task. For example, `jar` produces the consumable artifact for Java libraries. Attach tasks to this lifecycle task using `assemble.dependsOn(__task__)`.
+
+`build` — _lifecycle task_::
+(Depends on `assemble` and `check`) Intended to build everything, including running all tests, producing the production artifacts and generating documentation. You will probably rarely attach concrete tasks directly to `build` as `assemble` and `check` are typically more appropriate.
+
+`build__Configuration__` — task rule::
+Assembles those artifacts attached to the named configuration. For example, `buildArchives` will execute any task that is required to create any artifact attached to the `archives` configuration.
+
+`upload__Configuration__` — task rule::
+Does the same as `build__Configuration__`, but also uploads all the artifacts attached to the given configuration.
+
+`clean__Task__` — task rule::
+Removes the <<sec:task_inputs_outputs,defined outputs>> of a task, e.g. `cleanJar` will delete the JAR file produced by the `jar` task of the Java Plugin.
+
+
+[[sec:base_plugin_configurations]]
+=== Dependency management
+
+The Base Plugin adds no <<managing_dependency_configurations,configurations for dependencies>>, but it does add the following configurations for <<sec:artifacts_and_configurations,artifacts>>:
+
+`default`::
+By default, contains all the artifacts attached to all other configurations. It's used by consumer projects. For example, if project B has a `runtimeOnly` dependency on project A, then the artifacts in project A's `default` configuration are added to project B's `runtimeOnly` configuration.
```

Comment:
Thanks for checking the accuracy.

On the first point, I was going by this code in `BasePlugin`:

```
configurations.all(new Action<Configuration>() {
    public void execute(Configuration configuration) {
        configuration.getArtifacts().all(new Action<PublishArtifact>() {
            public void execute(PublishArtifact artifact) {
                defaultArtifacts.addCandidate(artifact);
            }
        });
    }
});
```

Looking further into it, `defaultArtifacts` only seems to be used if no artifacts are explicitly declared. And of course, the Java Plugin does do that with the `archives` configuration. Is that correct?

On the second point, was that behaviour added with variant-aware dependency resolution?

At this point, I'm wondering whether to mention the `default` configuration at all. On the one hand, it seems awfully confusing. On the other, this is the user manual and should be fairly comprehensive. And I think the configuration has a nasty way of rearing its head in real world builds. Or at least it did. Perhaps it's just not relevant any more.

Guidance would be much appreciated!

Reply:
The code you showed is about the `archives` configuration, not the `default` configuration.

Yes, the `default` configuration become essentially unused since the introduction of variant awareness. I think the proper description would be that "This is the configuration that will be used when no other configuration matches a dependency. It is only there for backwards compatibility and should not be used in new builds/plugins".